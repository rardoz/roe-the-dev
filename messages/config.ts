const localePrefix: 'as-needed' | 'always' | 'never' | undefined = 'as-needed'

const config = {
  locales: ['en-US', 'es-US'],
  defaultLocale: 'en-US',
  localePrefix,
}

export default config
