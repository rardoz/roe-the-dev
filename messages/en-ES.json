{
  "Home": {
    "title": "Roe the Dev. Hola!"
  }
}
