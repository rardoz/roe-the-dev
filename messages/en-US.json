{
  "Home": {
    "title": "Roe the Dev"
  },
  "Nav": {
    "portfolio": "Porfolio",
    "blog": "Blog",
    "experiments": "Experiments",
    "menuIconAltText": "Click for additional menu links",
    "logoAltText": "Roe the Dev logo",
    "about": "About Roe"
  },
  "About": {
    "title": "About Roe",
    "subtitle": "Tech Lead, Software Engineer, Mentor And Artist",
    "description1": "I am a versatile digital nomad and innovative creative technologist assuming the role of a tech lead while embracing a nomadic lifestyle, dedicated to bringing creative ideas to fruition.",
    "description2": "I pursued my education at Black Hawk College and The Illinois Institute of Art in Chicago, achieving an associate's degree in visual communications. With a professional tenure of over a decade, I have accumulated substantial experience in software development.",
    "cta": "Learn more"
  },
  "Portfolio": {
    "title": "Portfolio",
    "cta": "See my work",
    "footerMessage": "Background AI generated with"
  },
  "Blog": {
    "title": "From my desk",
    "cta": "Read blogs",
    "footerMessage": "Background AI generated with"
  }
}
