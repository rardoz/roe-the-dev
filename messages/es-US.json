{
  "Home": {
    "title": "Roe the Dev. Hola!"
  },
  "Nav": {
    "portfolio": "Cartera",
    "blog": "Blog",
    "experiments": "Experimentación",
    "menuIconAltText": "Haga clic para ver más enlaces",
    "logoAltText": "Logotipo de Roe the Dev",
    "about": "Sobre Roe"
  },
  "About": {
    "title": "Sobre Roe",
    "subtitle": "Líder Tecnológico, Ingeniero de Software, Mentor y Artista",
    "description1": "Soy un nómada digital versátil e innovador tecnólogo creativo, asumiendo el rol de líder tecnológico mientras abrazo un estilo de vida nómada, dedicado a llevar a cabo ideas creativas.",
    "description2": "Cursé mis estudios en Black Hawk College y en The Illinois Institute of Art en Chicago, obteniendo un título de asociado en comunicación visual. Con una experiencia profesional de más de una década, he acumulado una considerable experiencia en desarrollo de software.",
    "cta": "Aprender más"
  },
  "Portfolio": {
    "title": "Portafolio",
    "cta": "Ver mi trabajo",
    "footerMessage": "Fondo generado por inteligencia artificial con"
  },
  "Blog": {
    "title": "Desde mi escritorio",
    "cta": "Leer blogs"
  }
}
