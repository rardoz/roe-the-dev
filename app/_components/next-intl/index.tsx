import { createLocalizedPathnamesNavigation } from 'next-intl/navigation'

import config from '../../../messages/config'

export const { Link, redirect, usePathname, useRouter, getPathname } =
  createLocalizedPathnamesNavigation({
    locales: config.locales,
    pathnames: config.pathnames,
  })

import { ComponentProps } from 'react'
import type { AppPathnames } from '../../../messages/config'

export function UnstyledLink<Pathname extends AppPathnames>({
  href,
  ...rest
}: ComponentProps<typeof Link<Pathname | any>>) {
  return <Link href={href} {...rest} />
}

export function StyledLink<Pathname extends AppPathnames>({
  className,
  ...rest
}: ComponentProps<typeof Link<Pathname | any>>) {
  return (
    <UnstyledLink
      {...rest}
      className={`transition-colors duration-500 text-pink-500 hover:text-pink-400 ${
        className || ''
      }`}
    />
  )
}
