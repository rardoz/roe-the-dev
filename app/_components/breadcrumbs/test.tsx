import Breadcrumbs from '.'
import { render } from '@testing-library/react'
import { NextIntlClientProvider } from 'next-intl'
import messages from '../../../messages/en-US.json'
import { mockIntersectionObserver } from 'jsdom-testing-mocks'
mockIntersectionObserver()

describe('Breadcrumbs', () => {
  it('should render 1 crumb', () => {
    expect(
      render(
        <NextIntlClientProvider locale="en-US" messages={messages}>
          <Breadcrumbs links={[{ href: '#test1', label: 'test1' }]} />
        </NextIntlClientProvider>,
      ).container,
    ).toMatchSnapshot()
  })

  it('should render 2 crumb', () => {
    expect(
      render(
        <NextIntlClientProvider locale="en-US" messages={messages}>
          <Breadcrumbs
            links={[
              { href: '#test1', label: 'test1' },
              { href: '#test2', label: 'test2' },
            ]}
          />
        </NextIntlClientProvider>,
      ).container,
    ).toMatchSnapshot()
  })
})
