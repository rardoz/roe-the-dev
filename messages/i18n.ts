import { getRequestConfig } from 'next-intl/server'
import { default as config, locales } from './config'

// export default getRequestConfig(async ({ requestLocale }) => ({
//   messages: (await import(`./${requestLocale}.json`)).default,
// }))

export default getRequestConfig(async ({ requestLocale }) => {
  let _locale = await requestLocale
  if (!_locale || locales.includes(_locale as any)) {
    _locale = config.defaultLocale
  }

  return {
    locale: _locale,
    messages: (await import(`./${_locale}.json`)).default,
  }
})
