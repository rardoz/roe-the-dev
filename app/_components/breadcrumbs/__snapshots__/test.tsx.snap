// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumbs should render 1 crumb 1`] = `
<div>
  <div
    aria-label="Breadcrumb"
    class="flex"
  >
    <ol
      class="inline-flex flex-wrap items-start space-x-1 md:space-x-2 rtl:space-x-reverse"
    >
      <li
        class="inline-flex items-center"
      >
        <a
          class="transition-colors duration-500 text-pink-500 hover:text-pink-400 inline-flex items-center text-sm font-medium"
          href="/"
        >
          <svg
            class="w-3 h-3 me-1 text-pink-500"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 576 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M543.8 287.6c17 0 32-14 32-32.1c1-9-3-17-11-24L512 185V64c0-17.7-14.3-32-32-32H448c-17.7 0-32 14.3-32 32v36.7L309.5 7c-6-5-14-7-21-7s-15 1-22 8L10 231.5c-7 7-10 15-10 24c0 18 14 32.1 32 32.1h32v69.7c-.1 .9-.1 1.8-.1 2.8V472c0 22.1 17.9 40 40 40h16c1.2 0 2.4-.1 3.6-.2c1.5 .1 3 .2 4.5 .2H160h24c22.1 0 40-17.9 40-40V448 384c0-17.7 14.3-32 32-32h64c17.7 0 32 14.3 32 32v64 24c0 22.1 17.9 40 40 40h24 32.5c1.4 0 2.8 0 4.2-.1c1.1 .1 2.2 .1 3.3 .1h16c22.1 0 40-17.9 40-40V455.8c.3-2.6 .5-5.3 .5-8.1l-.7-160.2h32z"
            />
          </svg>
          Home
        </a>
      </li>
      <li
        class="inline-flex items-center me-1"
      >
        <svg
          class="me-1 w-3 h-3 text-pink-500"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 320 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M310.6 233.4c12.5 12.5 12.5 32.8 0 45.3l-192 192c-12.5 12.5-32.8 12.5-45.3 0s-12.5-32.8 0-45.3L242.7 256 73.4 86.6c-12.5-12.5-12.5-32.8 0-45.3s32.8-12.5 45.3 0l192 192z"
          />
        </svg>
        <span
          class="text-sm font-medium text-purple-500"
        >
          test1
        </span>
      </li>
    </ol>
  </div>
</div>
`;

exports[`Breadcrumbs should render 2 crumb 1`] = `
<div>
  <div
    aria-label="Breadcrumb"
    class="flex"
  >
    <ol
      class="inline-flex flex-wrap items-start space-x-1 md:space-x-2 rtl:space-x-reverse"
    >
      <li
        class="inline-flex items-center"
      >
        <a
          class="transition-colors duration-500 text-pink-500 hover:text-pink-400 inline-flex items-center text-sm font-medium"
          href="/"
        >
          <svg
            class="w-3 h-3 me-1 text-pink-500"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 576 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M543.8 287.6c17 0 32-14 32-32.1c1-9-3-17-11-24L512 185V64c0-17.7-14.3-32-32-32H448c-17.7 0-32 14.3-32 32v36.7L309.5 7c-6-5-14-7-21-7s-15 1-22 8L10 231.5c-7 7-10 15-10 24c0 18 14 32.1 32 32.1h32v69.7c-.1 .9-.1 1.8-.1 2.8V472c0 22.1 17.9 40 40 40h16c1.2 0 2.4-.1 3.6-.2c1.5 .1 3 .2 4.5 .2H160h24c22.1 0 40-17.9 40-40V448 384c0-17.7 14.3-32 32-32h64c17.7 0 32 14.3 32 32v64 24c0 22.1 17.9 40 40 40h24 32.5c1.4 0 2.8 0 4.2-.1c1.1 .1 2.2 .1 3.3 .1h16c22.1 0 40-17.9 40-40V455.8c.3-2.6 .5-5.3 .5-8.1l-.7-160.2h32z"
            />
          </svg>
          Home
        </a>
      </li>
      <li
        class="inline-flex items-center me-1"
      >
        <svg
          class="me-1 w-3 h-3 text-pink-500"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 320 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M310.6 233.4c12.5 12.5 12.5 32.8 0 45.3l-192 192c-12.5 12.5-32.8 12.5-45.3 0s-12.5-32.8 0-45.3L242.7 256 73.4 86.6c-12.5-12.5-12.5-32.8 0-45.3s32.8-12.5 45.3 0l192 192z"
          />
        </svg>
        <a
          class="transition-colors duration-500 text-pink-500 hover:text-pink-400 text-sm font-medium"
          href="#test1"
        >
          test1
        </a>
      </li>
      <li
        class="inline-flex items-center me-1"
      >
        <svg
          class="me-1 w-3 h-3 text-pink-500"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 320 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M310.6 233.4c12.5 12.5 12.5 32.8 0 45.3l-192 192c-12.5 12.5-32.8 12.5-45.3 0s-12.5-32.8 0-45.3L242.7 256 73.4 86.6c-12.5-12.5-12.5-32.8 0-45.3s32.8-12.5 45.3 0l192 192z"
          />
        </svg>
        <span
          class="text-sm font-medium text-purple-500"
        >
          test2
        </span>
      </li>
    </ol>
  </div>
</div>
`;
